catroid/src/main/java/org/catrobat/catroid/formulaeditor/datacontainer/DataContainer.java

```diff
@@ -24,350 +24,246 @@
 
 import android.content.Context;
 
-import org.catrobat.catroid.ProjectManager;
-import org.catrobat.catroid.R;
 import org.catrobat.catroid.content.Project;
 import org.catrobat.catroid.content.Sprite;
 import org.catrobat.catroid.content.bricks.UserBrick;
 import org.catrobat.catroid.formulaeditor.UserList;
 import org.catrobat.catroid.formulaeditor.UserVariable;
-import org.catrobat.catroid.ui.UserBrickSpriteActivity;
 import org.catrobat.catroid.ui.adapter.DataAdapter;
 
-import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
-import java.util.LinkedList;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 
 public class DataContainer extends BaseDataContainer {
 
-	public enum DataType implements Serializable {
-		USER_VARIABLE_SPRITE,
-		USER_VARIABLE_PROJECT,
-		USER_VARIABLE_USERBRICK,
-		USER_LIST_SPRITE,
-		USER_LIST_PROJECT,
-		USER_DATA_EMPTY
-	}
-
-	private transient Project project;
+	private transient List<UserVariable> projectVariables = new ArrayList<>();
+	private transient List<UserList> projectLists = new ArrayList<>();
 
-	private transient SpriteVariableBehaviour spriteVariableBehaviour = new SpriteVariableBehaviour(this);
-	private transient SpriteListBehaviour spriteListBehaviour = new SpriteListBehaviour(this);
-	private transient UserBrickVariableBehaviour userBrickVariableBehaviour = new UserBrickVariableBehaviour(this);
-	private transient ProjectVariableBehaviour projectVariableBehaviour = new ProjectVariableBehaviour(this);
-	private transient ProjectListBehaviour projectListBehaviour = new ProjectListBehaviour(this);
-
-	private DataContainer() {
+	public DataContainer() {
+		spriteVariables = new HashMap<>();
+		spriteListOfLists = new HashMap<>();
```

Comment:
Since you're assigning new `HashMap`s in all constructors, you could also move these initializations to the `BaseDataContainer`.

If you're interested in how xstream handles members during deserializing, [here is a gist](https://gist.github.com/robertpainsi/a34c1934024ffe667b41) I made last year for testing.

Reply:
I think we don't need the BaseDataContainer anymore anyway. However, I need to fix de-serialization for "old" projects anyway. We'll see how that goes. 😉